modules/core/src/matrix.cpp

```diff
@@ -1261,6 +1261,16 @@ Mat Mat::reshape(int _cn, const std::vector<int>& _newshape) const
     return reshape(_cn, (int)_newshape.size(), &_newshape[0]);
 }
 
+Mat Mat::reinterpretType(int type) const
+{
+    type = CV_MAT_TYPE(type);
+    CV_Assert(CV_ELEM_SIZE(this->type()) == CV_ELEM_SIZE(type));
+    Mat m = *this;
+    m.flags = (type & CV_MAT_TYPE_MASK) | MAGIC_VAL;
```

Comment:
it's safer to use `m.flags = (m.flags & ~CV_MAT_TYPE_MASK) | type;` here.

Reply:
@vpisarev I had done